Articles/Blog/WhenIsBetterToWriteAutomatedTests-src.md

```diff
@@ -0,0 +1,112 @@
+# DRAFT: When is it Better not to Write Automated Tests?
+
+**Hero Image:**
+
+- <img src='https://github.com/betterscientificsoftware/images/raw/master/testing-hero-image.png'/>[Image Source: https://www.testingspot.net (see below)]
+
+#### Contributed by [Roscoe A. Barltett](https://bartlettroscoe.github.io)
+
+#### Publication date: MONTH DD, YEAR
+
+<br>
+
+## Introduction
+
+The importance of writing and maintaining strong automated tests for software is well established in the modern software engineering community (and in fact, tests will be written before the code when using test-driven development, TDD).
+But does it always pay off to write automated tests for some piece of code?
+Are there situations where one is better off not writing automated tests?
+Will you be considered to be a bad developer and chastised by your peers, users, stakeholders, or your manager if every line of code that you write is not under strong automated testing?
+
+I have been wrestling with this question since 2007 when I first read the book [*Working Effectively with Legacy Code*](https://bssw.io/items/working-effectively-with-legacy-code/)<sup>[1]</sup> and radically changed my thinking about automated testing.
+In the 10+ years of my professional career prior to 2007, I wrote a lot of Computational Science and Engineering (CSE) and applied math software with very little strong automated testing.
+I used the validation-centric approach<sup>[2]</sup> to test the software that I was writing (i.e. only test/validate the software in the downstream customer code and then perhaps only manually).
+As a result, I had personally written O(100k) lines of code that were being used by other people and was in the position to have to maintain.
+When I realized that almost all of this software was considered "Legacy Code" according to *Working Effectively with Legacy Code* (i.e. "code without tests == legacy code"), I felt ashamed and horrified at the mass of useful but buggy and difficult-to=maintain software that I had created and now would have to deal with for years to come.
```

Comment:
did you mean difficult-to-maintain?

Reply:
@prwolfe asked:

> did you mean difficult-to-maintain?

Yea, that was a typo.  It is fixed in https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/pull/377/commits/83e39ea3cb454e693c78e1d812877d3b2b035626#diff-ff08f44603f4e10f3095baf6cea0da6eR24.

Reply:
Thanks Ross!